server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -244,6 +251,24 @@ public MailImpl build() {
             mail.perRecipientSpecificHeaders.addAll(perRecipientHeaders);
             return mail;
         }
+
+        private Map<String, Serializable> toAttributes(Map<String, ByteBuffer> rowAttributes) {
+            return rowAttributes.entrySet()
+                .stream()
+                .map(entry -> Pair.of(entry.getKey(), fromByteBuffer(entry.getValue())))
```

Comment:
Why MailImpl has to deal with byte buffers?

I guess you are trying to share code between here and CassandramailRepository but it do not sound like a good idea to me to put it here...

Reply:
I will revert this